tests/unit/waveform/test_digital_waveform_signal.py

```diff
@@ -0,0 +1,262 @@
+    "left, right",
+    [
+        (
+            DigitalWaveform.from_lines([0, 1, 2, 3], np.uint8).signals[0],
+            DigitalWaveform.from_lines([0, 1, 2, 3], np.uint8).signals[0],
+        ),
+        (
+            DigitalWaveform.from_lines([False, True, False], np.bool).signals[0],
+            DigitalWaveform.from_lines([False, True, False], np.bool).signals[0],
+        ),
+        # Equality does not take the signal index or signal name into account.
+        (
+            DigitalWaveform(3, 2, extended_properties={"NI_LineNames": "0, 1"}).signals[0],
+            DigitalWaveform(3, 2, extended_properties={"NI_LineNames": "0, 1"}).signals[1],
+        ),
+    ],
+)
+def test___same_value___equality___equal(
+    left: DigitalWaveformSignal[Any], right: DigitalWaveformSignal[Any]
+) -> None:
+    assert left == right
+    assert not (left != right)
+
+
+@pytest.mark.parametrize(
+    "left, right",
+    [
+        (
+            DigitalWaveform.from_lines([0, 1, 2, 3], np.uint8).signals[0],
+            DigitalWaveform.from_lines([0, 1, 4, 3], np.uint8).signals[0],
+        ),
+        (
+            DigitalWaveform.from_lines([False, True, False], np.bool).signals[0],
+            DigitalWaveform.from_lines([False, False, False], np.bool).signals[0],
+        ),
+    ],
+)
+def test___different_value___equality___not_equal(
+    left: DigitalWaveformSignal[Any], right: DigitalWaveformSignal[Any]
+) -> None:
+    assert not (left == right)
+    assert left != right
+
+
+@pytest.mark.parametrize(
+    "value, expected_repr",
+    [
+        (
+            DigitalWaveform(3, 2).signals[0],
+            "nitypes.waveform.DigitalWaveformSignal(data=array([0, 0, 0], dtype=uint8))",
+        ),
+        (
+            DigitalWaveform(3, 2, np.bool).signals[0],
+            "nitypes.waveform.DigitalWaveformSignal(data=array([False, False, False]))",
+        ),
+        (
+            DigitalWaveform(
+                3, 2, extended_properties={"NI_LineNames": "port0/line0, port0/line1"}
+            ).signals[1],
+            "nitypes.waveform.DigitalWaveformSignal(name='port0/line1', data=array([0, 0, 0], dtype=uint8))",
+        ),
+    ],
+)
+def test___various_values___repr___looks_ok(
+    value: DigitalWaveformSignal[Any], expected_repr: str
+) -> None:
+    assert repr(value) == expected_repr
+
+
+_VARIOUS_VALUES = [
+    DigitalWaveform(3, 2).signals[0],
+    DigitalWaveform(3, 2, np.bool).signals[0],
+    DigitalWaveform(3, 2, extended_properties={"NI_LineNames": "port0/line0, port0/line1"}).signals[
+        1
+    ],
+]
+
+
+@pytest.mark.parametrize("value", _VARIOUS_VALUES)
+def test___various_values___copy___makes_shallow_copy(value: DigitalWaveformSignal[Any]) -> None:
+    new_value = copy.copy(value)
+
+    _assert_shallow_copy(new_value, value)
+
+
+def _assert_shallow_copy(
+    value: DigitalWaveformSignal[Any], other: DigitalWaveformSignal[Any]
+) -> None:
+    assert value == other
+    assert value is not other
+
```

Comment:
Oops, the distinction between shallow vs. deep copies for this class is whether it copies _owner